Makefile

```diff
@@ -31,8 +31,10 @@ generate-accounts: ALWAYS_RUN
 # `docker ps`:  Remove any lingering test containers
 # `.gitignore`: Remove `test/duplicate_configs` and files copied via `make generate-accounts`
 clean: ALWAYS_RUN
-	-@ for CONTAINER in $$(docker ps -a --filter name='^dms-test_.*|^mail_.*|^hadolint$$|^eclint$$|^shellcheck$$' | sed 1d | cut -f 1-1 -d ' '); do docker rm -f $${CONTAINER}; done
-	-@ while read -r LINE; do [[ $${LINE} =~ test/.+ ]] && sudo rm -rf $${LINE}; done < .gitignore
+	-@ while read -r LINE; do CONTAINERS+=("$${LINE}"); done < <(docker ps -qaf name='^(dms-test|mail)_.*') ; \
+	for CONTAINER in "$${CONTAINERS[@]}"; do docker rm -f "$${CONTAINER}"; done
+	-@ while read -r LINE; do [[ $${LINE} =~ test/.+ ]] && FILES+=("/mnt$${LINE#test}"); done < .gitignore ; \
+	docker run --rm -v "$(REPOSITORY_ROOT)/test/:/mnt" docker.io/alpine:3.17.1 ash -c "rm -rf $${FILES[@]}"
```

Comment:
```suggestion
	-@ while read -r LINE; do \
		CONTAINERS+=("$${LINE}"); \
	done < <(docker ps -qaf name='^(dms-test|mail)_.*') ; \
	for CONTAINER in "$${CONTAINERS[@]}"; do \
		docker rm -f "$${CONTAINER}"; \
	done
	-@ while read -r LINE; do \
		[[ $${LINE} =~ test/.+ ]] && \
		FILES+=("/mnt$${LINE#test}"); \
	done < .gitignore ; \
	docker run --rm -v "$(REPOSITORY_ROOT)/test/:/mnt" docker.io/alpine:3.17.1 ash -c "rm -rf $${FILES[@]}"
```

Reply:
At this point I'm wondering if it should just be a small script that gets called outside of the Makefile :sweat_smile: 

I don't know about pinning the alpine version, the minor changes every few months, we're not likely to keep that maintained, especially with the point releases. Perhaps just omit the tag for implicit `:latest`? Shouldn't cause any issues for this usage.

Reply:
Good point about the version pinning. I don't think the behaviour of `rm` will significantly change with newer versions 😆 Sticking to `:latest` should be fine.

Reply:
Definitely keep it in the Makefile, no outside script. And the version pinning is just so Docker does not pull a new version ever time the image is updated.

Reply:
> Docker does not pull a new version ever time the image is updated.

It shouldn't for `:latest`, that's often a drawback due to newer Docker users assuming it always gets the `:latest` rather than the latest digest at the time it was originally pulled, then uses the local `:latest` tagged image from then on (_unless explicitly pulling again_).

Reply:
I'm not overly fussed either way due to usage, but the version pinning here isn't serving much purpose.

I know in our tests we have a very outdated version pinning for OpenLDAP container that I'll sort out when I get around to working on the LDAP test.

Our CI may want to bump from Ubuntu 20.04 LTS to 22.04 LTS at some point, but no rush there as it shouldn't offer any advantage to us I think. Github keeps the pre-installed software we care about updated (eg Docker Engine).

Reply:
There you go: [8f006ba](https://github.com/docker-mailserver/docker-mailserver/pull/3020/commits/8f006bab40371fb7e200b9934d5854c82a2b5377)

Can we please get over with this?